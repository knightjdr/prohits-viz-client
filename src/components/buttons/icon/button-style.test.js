import React from 'react';
import { render } from '@testing-library/react';

import Button from './button-style';

const theme = {
  colorPrimary1: '#0000ff',
  colorSecondary1: '#00ff00',
  fontLight: '#fff',
  fontDark: '#111',
  warning1: '#ee000',
  warning2: '#ff000',
};

describe('Icon button style', () => {
  describe('default', () => {
    let container;

    beforeAll(() => {
      ({ container } = render(<Button theme={theme} />));
    });

    it('should match snapshot', () => {
      expect(container).toMatchSnapshot();
    });
  });

  describe('primary', () => {
    let container;

    beforeAll(() => {
      ({ container } = render(
        <Button
          kind="primary"
          theme={theme}
        />,
      ));
    });

    it('should match snapshot', () => {
      expect(container).toMatchSnapshot();
    });

    it('should set background color', () => {
      expect(container.firstChild).toHaveStyleRule('background-color', theme.colorPrimary1);
    });

    it('should set color', () => {
      expect(container.firstChild).toHaveStyleRule('color', theme.fontLight);
    });

    it('should set icon background color on focus', () => {
      const modifier = {
        modifier: '&:focus:not([disabled])',
      };
      expect(container.firstChild).toHaveStyleRule('background-color', theme.accentPrimary1, modifier);
    });

    it('should set icon color on focus', () => {
      const modifier = {
        modifier: '&:focus:not([disabled])',
      };
      expect(container.firstChild).toHaveStyleRule('color', theme.colorPrimary1, modifier);
    });

    it('should set icon background color on hover', () => {
      const modifier = {
        modifier: '&:hover:not([disabled])',
      };
      expect(container.firstChild).toHaveStyleRule('background-color', theme.accentPrimary1, modifier);
    });

    it('should set icon color on hover', () => {
      const modifier = {
        modifier: '&:hover:not([disabled])',
      };
      expect(container.firstChild).toHaveStyleRule('color', theme.colorPrimary1, modifier);
    });
  });

  describe('secondary', () => {
    let container;

    beforeAll(() => {
      ({ container } = render(
        <Button
          kind="secondary"
          theme={theme}
        />,
      ));
    });

    it('should match snapshot', () => {
      expect(container).toMatchSnapshot();
    });

    it('should set background color', () => {
      expect(container.firstChild).toHaveStyleRule('background-color', theme.colorSecondary1);
    });

    it('should set color', () => {
      expect(container.firstChild).toHaveStyleRule('color', theme.fontLight);
    });

    it('should set filter on focus', () => {
      const modifier = {
        modifier: '&:focus:not([disabled])',
      };
      expect(container.firstChild).toHaveStyleRule('filter', 'contrast(200%)', modifier);
    });

    it('should set filter on hover', () => {
      const modifier = {
        modifier: '&:hover:not([disabled])',
      };
      expect(container.firstChild).toHaveStyleRule('filter', 'contrast(200%)', modifier);
    });
  });

  describe('transparent', () => {
    let container;

    beforeAll(() => {
      ({ container } = render(<Button theme={theme} />));
    });

    it('should match snapshot', () => {
      expect(container).toMatchSnapshot();
    });

    it('should set background color', () => {
      expect(container.firstChild).toHaveStyleRule('background-color', 'transparent');
    });

    it('should set color', () => {
      expect(container.firstChild).toHaveStyleRule('color', theme.colorPrimary1);
    });

    it('should set before background color', () => {
      const modifier = {
        modifier: '&::before',
      };
      expect(container.firstChild).toHaveStyleRule('background-color', theme.colorPrimary1, modifier);
    });

    it('should set icon color on focus', () => {
      const modifier = {
        modifier: '&:focus:not([disabled])',
      };
      expect(container.firstChild).toHaveStyleRule('color', theme.fontLight, modifier);
    });

    it('should set icon color on hover', () => {
      const modifier = {
        modifier: '&:hover:not([disabled])',
      };
      expect(container.firstChild).toHaveStyleRule('color', theme.fontLight, modifier);
    });
  });

  describe('warning', () => {
    let container;

    beforeAll(() => {
      ({ container } = render(
        <Button
          kind="warning"
          theme={theme}
        />,
      ));
    });

    it('should match snapshot', () => {
      expect(container).toMatchSnapshot();
    });

    it('should set background color', () => {
      expect(container.firstChild).toHaveStyleRule('background-color', theme.warning2);
    });

    it('should set color', () => {
      expect(container.firstChild).toHaveStyleRule('color', theme.fontDark);
    });

    it('should set icon background color on focus', () => {
      const modifier = {
        modifier: '&:focus:not([disabled])',
      };
      expect(container.firstChild).toHaveStyleRule('background-color', theme.warning1, modifier);
    });

    it('should set icon color on focus', () => {
      const modifier = {
        modifier: '&:focus:not([disabled])',
      };
      expect(container.firstChild).toHaveStyleRule('color', theme.fontLight, modifier);
    });

    it('should set icon background color on hover', () => {
      const modifier = {
        modifier: '&:hover:not([disabled])',
      };
      expect(container.firstChild).toHaveStyleRule('background-color', theme.warning1, modifier);
    });

    it('should set icon color on hover', () => {
      const modifier = {
        modifier: '&:hover:not([disabled])',
      };
      expect(container.firstChild).toHaveStyleRule('color', theme.fontLight, modifier);
    });
  });
});
