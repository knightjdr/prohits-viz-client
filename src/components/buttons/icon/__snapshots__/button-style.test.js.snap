// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon button style default should match snapshot 1`] = `
.c0 {
  background-color: transparent;
  border-color: transparent;
  box-sizing: border-box;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-width: 2px;
  color: #0000ff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.3em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 34px;
  padding: 2px;
  position: relative;
  width: 34px;
}

.c0:focus {
  outline: none;
}

.c0::-moz-focus-inner {
  border: 0;
}

.c0:focus,
.c0:hover {
  box-shadow: 0 2px 5px 0 rgba(0,0,0,0.26);
}

.c0:active {
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-name: click;
  animation-name: click;
}

.c0::before {
  border-radius: 50%;
  content: '';
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  -webkit-transition: inherit;
  transition: inherit;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transform-origin: center center;
  -ms-transform-origin: center center;
  transform-origin: center center;
  width: 100%;
  z-index: -1;
}

.c0 {
  background-color: transparent;
  border: none;
  border-radius: 50%;
}

.c0::before {
  background-color: #0000ff;
  opacity: 0;
}

.c0:focus:not([disabled]),
.c0:hover:not([disabled]) {
  box-shadow: none;
  color: #fff;
}

.c0:focus:not([disabled])::before,
.c0:hover:not([disabled])::before {
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

<div>
  <button
    class="c0"
    kind="transparent"
  />
</div>
`;

exports[`Icon button style transparent should match snapshot 1`] = `
.c0 {
  background-color: transparent;
  border-color: transparent;
  box-sizing: border-box;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-width: 2px;
  color: #0000ff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.3em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 34px;
  padding: 2px;
  position: relative;
  width: 34px;
}

.c0:focus {
  outline: none;
}

.c0::-moz-focus-inner {
  border: 0;
}

.c0:focus,
.c0:hover {
  box-shadow: 0 2px 5px 0 rgba(0,0,0,0.26);
}

.c0:active {
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-name: click;
  animation-name: click;
}

.c0::before {
  border-radius: 50%;
  content: '';
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  -webkit-transition: inherit;
  transition: inherit;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transform-origin: center center;
  -ms-transform-origin: center center;
  transform-origin: center center;
  width: 100%;
  z-index: -1;
}

.c0 {
  background-color: transparent;
  border: none;
  border-radius: 50%;
}

.c0::before {
  background-color: #0000ff;
  opacity: 0;
}

.c0:focus:not([disabled]),
.c0:hover:not([disabled]) {
  box-shadow: none;
  color: #fff;
}

.c0:focus:not([disabled])::before,
.c0:hover:not([disabled])::before {
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

<div>
  <button
    class="c0"
    kind="transparent"
  />
</div>
`;
